src/main/java/com/mongodb/kafka/connect/util/ConnectionValidator.java

```diff
@@ -182,26 +168,65 @@ public static void validateUserHasActions(
                           missingPermissions)));
     } catch (MongoSecurityException e) {
       ConfigHelper.getConfigByName(config, configName)
-          .ifPresent(c -> c.addErrorMessage("Invalid user permissions authentication failed."));
+          .ifPresent(
+              c ->
+                  c.addErrorMessage(
+                      "Invalid user permissions authentication failed. " + e.getMessage()));
     } catch (Exception e) {
       LOGGER.warn("Permission validation failed due to: {}", e.getMessage(), e);
     }
   }
 
+  private static List<String> validateRoleHasActions(
```

Comment:
This method behaves similarly to `removeUserActions`: it gets a set of actions and returns a subset representing those actions that are not permitted. However, the naming is different, which negatively contributes to readability.

Another thing I noticed is that the method `removeUserActions` does not modify the passed `actions` collection, which yet its name implies modification.

Can we name this methods in a more consistent and descriptive way?

Reply:
Will do